Kubernetes/HOWTO-on-prem.md

```diff
@@ -0,0 +1,103 @@
+
+### Install CNI ###
+We also need to install the basic CNI binaries so that networking works. Download them from [here](https://github.com/containernetworking/plugins/releases) and copy them to `/opt/cni/bin/`.
+
+## Prepare the Master ##
+Copy the entire directory from [this repository](https://github.com/Microsoft/SDN/tree/master/Kubernetes/linux), which contains a lot of the setup scripts and Kubernetes files that are essential to this process. Check them all out to `~/kube/` and make the scripts executable. This entire directory will be getting mounted for a lot of the docker containers in future steps, so keep its structure the same as outlined in the guide. 
+
+We'll be creating the service cluster on the virtual subnet 11.0/16. This isn't configurable, so if you want to tweak this, it'll require manual modification.
+
+### Certificates ###
+First, prepare the certificates that will be used for nodes to communicate inthe cluster. Run `generate-certs.sh` (in the `certs/` directory), passing **your** external IP as its (only) parameter -- this is likely whatever the IP address of your `eth0` interface, or whatever:
+
+    ~/kube/certs $ ./generate-certs.sh 10.123.45.67   # example! replace
+
+### Allow Routing ###
+_This step may be optional, depending on whether or not you've configured your intended cluster subnet to be routable already_.
+
+Windows nodes will each grab a /24 from the cluster, so if your cluster CIDR is 192.168.0.0/16, a Windows node will use 192.168.0.1/24. Just run:
+
+    ~/kube $ generate-routes.sh 192.168
+
+### Prepare Manifests & Addons ###
+In the `manifest` folder, run the Python script, passing your master IP and the _full_ cluster CIDR:
+
+    $ python2 generate.py 10.123.45.67 192.168.0.0/16
+
+This will generate a set of YAML files. You should [re]move the Python script so that K8s doesn't mistake it for a manifest and cause problems.
+
+### Configure & Run Kubernetes ###
+Run the script `configure-kubectl.sh`, passing your external IP again, to create a configuration in `~/.kube/config` containing all of the certificate data and other parameters:
+
+    ~/kube $ ./configure-kubectl.sh 10.123.45.67
+
+Run the script `start-kubelet.sh`. You may need to run it with `sudo`. Then, in another shell, copy the Kubernetes configuration file to the local kubelet mount path:
+
+    $ sudo cp ~/.kube/config kubelet/
+
+**TODO**: Why is this necessary? Shouldn't this be done automatically by one of the containers?
+
+In another terminal session, run the Kubeproxy script, passing your cluster CIDR (you may also need to run this one with `sudo`):
+
+    ~/kube $ ./kubeproxy.sh 192.168
+
+After some time, you should see the following system state:

```

Comment:
"After some time" - people get impatient quickly (I do!) so do we know a rough timeframe? Are we talking 2 minutes or should folks wait ~15 minutes before checking for the system state you mention.

Reply:
The error I keep seeing is

E1009 09:15:05.072900   94524 reflector.go:205] k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/factory.go:73: Failed to list *api.Service: Get https://10.127.132.211/api/v1/services?resourceVersion=0: dial tcp 10.127.132.211:443: getsockopt: connection refused

is this expected?

Reply:
If ApiServer is not running, then it might be due to the issue I mentioned in generate.py (whitespace before "apiVersion" in the files generated by generate.py. That was causing my ApiServer to not come up.

Reply:
Should be about a minute -- will clarify.